Source/WebCore/PAL/pal/text/TextEncodingRegistry.cpp

```diff
@@ -81,77 +82,89 @@ struct TextEncodingNameHash {
                 return h;
             }
             h += toASCIILower(c);
-            h += (h << 10); 
-            h ^= (h >> 6); 
+            h += (h << 10);
+            h ^= (h >> 6);
         }
     }
 
     static const bool safeToCompareToEmptyOrDeleted = false;
 };
 
-using TextEncodingNameMap = HashMap<const char*, const char*, TextEncodingNameHash>;
-using TextCodecMap = HashMap<const char*, NewTextCodecFunction>;
+struct HashTranslatorTextEncodingName {
+    static unsigned hash(const char* literal)
+    {
+        return TextEncodingNameHash::hash(literal);
+    }
+
+    static bool equal(const ASCIILiteral& a, const char* b)
+    {
+        return TextEncodingNameHash::equal(a.characters(), b);
+    }
+};
+
+using TextEncodingNameMap = HashMap<ASCIILiteral, ASCIILiteral, TextEncodingNameHash>;
+using TextCodecMap = HashMap<ASCIILiteral, NewTextCodecFunction>;
 
 static Lock encodingRegistryLock;
 
 static TextEncodingNameMap* textEncodingNameMap WTF_GUARDED_BY_LOCK(encodingRegistryLock);
 static TextCodecMap* textCodecMap WTF_GUARDED_BY_LOCK(encodingRegistryLock);
 static bool didExtendTextCodecMaps;
-static HashSet<const char*>* japaneseEncodings;
-static HashSet<const char*>* nonBackslashEncodings;
+static HashSet<ASCIILiteral>* japaneseEncodings;
+static HashSet<ASCIILiteral>* nonBackslashEncodings;
 
-static const char* const textEncodingNameBlocklist[] = { "UTF-7", "BOCU-1", "SCSU" };
+static constexpr ASCIILiteral textEncodingNameBlocklist[] = { "UTF-7"_s, "BOCU-1"_s, "SCSU"_s };
 
-static bool isUndesiredAlias(const char* alias)
+static bool isUndesiredAlias(ASCIILiteral alias)
 {
     // Reject aliases with version numbers that are supported by some back-ends (such as "ISO_2022,locale=ja,version=0" in ICU).
-    for (const char* p = alias; *p; ++p) {
-        if (*p == ',')
+    for (unsigned i = 0; i < alias.length(); ++i) {
+        if (alias.characterAt(i) == ',')
```

Comment:
Surprised we have to write out a loop just to check for the presence of a character. Seems like `strchr` and `contains`.

Reply:
Switching to `strchr()`.